template-files/action.py

```diff
@@ -0,0 +1,217 @@
+                dst = None if (tmp := file.get("dst", src)) is None else Path(tmp)
+                remove = file.get("remove", False)
+                context = file.get("with", {})
+            else:
+                perror(f"❌ Invalid file definition ({file}), expected str or dict")
+                errors += 1
+                continue
+
+            if remove:
+                if dst is None:
+                    perror(f"❌ Invalid file definition ({file}), expected dst")
+                    errors += 1
+                    continue
+
+                try:
+                    dst.unlink()
+                except FileNotFoundError:
+                    # FileNotFoundError: dst does not exist
+                    print(f"⚠️ {dst} has already been removed")
+                except PermissionError as err:
+                    # PermissionError: not possible to remove dst
+                    perror(f"❌ Failed to remove {dst}: {err}")
+                    errors += 1
+                else:
+                    print(f"✅ Removed {dst}")
+                continue
+
+            try:
+                content = destination.get_contents(src).decoded_content.decode()
+            except UnknownObjectException as err:
+                perror(f"❌ Failed to fetch {src} from {repository}: {err}")
+                errors += 1
+                continue
+
+            # inject stuff about the source and destination
+            context["repo"] = context["dst"] = context["destination"] = source
+            context["src"] = context["source"] = destination
+
+            template = env.from_string(content)
+            dst.parent.mkdir(parents=True, exist_ok=True)
+            dst.write_text(template.render(**context))
+
+            print(f"✅ Templated {repository}/{src} as {dst}")
+
+    return errors
+
+
+def main():
+    errors = 0
+
+    args = parse_args()
+    if not args.config:
+        print("⚠️ No configuration file found, nothing to update")
+        sys.exit(0)
+
+    config = read_config(args)
+
+    # initialize Jinja environment and GitHub client
+    env = Environment(
+        loader=FileSystemLoader(args.stubs),
+        # {{ }} is used in MermaidJS
+        # ${{ }} is used in GitHub Actions
+        # { } is used in Python
+        # %( )s is used in Python
+        block_start_string="[%",
+        block_end_string="%]",
+        variable_start_string="[[",
+        variable_end_string="]]",
+        comment_start_string="[#",
+        comment_end_string="#]",
+        keep_trailing_newline=True,
+    )
+    gh = Github(auth=Auth.Token(os.environ["GITHUB_TOKEN"]))
+
+    # get current repository
+    repository = os.environ["GITHUB_REPOSITORY"]
+    try:
+        source = gh.get_repo(repository)
+    except UnknownObjectException as err:
+        perror(f"❌ Failed to fetch {repository}: {err}")
+        errors += 1
+
+    if not errors:
+        errors += iterate_config(config, gh, env, source)
```

Comment:
This is wrongly named, the GITHUB_REPOSITORY repo is the one where the action is being executed in and files copied to, so not the "source" of the files, but the "destination" of files? A little confusion, maybe `context_repo` might make more sense?

```suggestion
    # get current repository
    try:
        context_repo = gh.get_repo(os.environ["GITHUB_REPOSITORY"])
    except UnknownObjectException as err:
        perror(f"❌ Failed to fetch {repository}: {err}")
        errors += 1

    if not errors:
        errors += iterate_config(config, gh, env, context_repo)
```

Reply:
yeah you're right, I had them flipped

`context_repo` IMO is probably just as confusing since we have a `context` variable as well

I'm currently experimenting with `current_repo` and `upstream_repo` for the variables

Reply:
Sure, that works for me